src/common/src/hash/consistent_hash/vnode.rs

```diff
@@ -119,9 +119,10 @@ impl VirtualNode {
         if let Ok(idx) = keys.iter().exactly_one()
             && let ArrayImpl::Serial(serial_array) = &**data_chunk.column_at(*idx)
         {
+            let fallback_row_id = Serial::from(rand::random::<i64>());
```

Comment:
Why do we need this? And also for `compute_row`?

Reply:
This looks too hack to me. 🥵 It could be surprising that this is not a pure function. Since the motivation here is to avoid data skew, I guess using the hash code of the row is way much better as it guarantees stability at least.

Can we add another `RowIdGen` here as well? We just need to ensure that there's no overlapping with the latter one, which can be done by allocating a flag of one bit for this information I guess.

Reply:
> Can we add another RowIdGen here as well? We just need to ensure that there's no overlapping with the latter one, which can be done by allocating a flag of one bit for this information I guess.

A little strange to me because we can not know which data source can give more records